module_ctx.go

```diff
@@ -775,17 +775,17 @@ type TopDownMutatorContext interface {
 type BottomUpMutatorContext interface {
 	BaseMutatorContext
 
-	// AddDependency adds a dependency to the given module.
-	// Does not affect the ordering of the current mutator pass, but will be ordered
-	// correctly for all future mutator passes.
-	AddDependency(module Module, tag DependencyTag, name ...string)
-
-	// AddReverseDependency adds a dependency from the destination to the given module.
-	// Does not affect the ordering of the current mutator pass, but will be ordered
-	// correctly for all future mutator passes.  All reverse dependencies for a destination module are
-	// collected until the end of the mutator pass, sorted by name, and then appended to the destination
-	// module's dependency list.
-	AddReverseDependency(module Module, tag DependencyTag, name string)
+	// AddDependency adds a dependency to the given module.  It returns a slice of modules for each
+	// dependency (some entries may be nil).  Does not affect the ordering of the current mutator
+	// pass, but will be ordered correctly for all future mutator passes.
+	AddDependency(module Module, tag DependencyTag, name ...string) []Module
+
+	// AddReverseDependency adds a dependency from the destination to the given module.  It returns
+	// a module for reverse dependency.  Does not affect the ordering of the current mutator pass,
+	// but will be ordered correctly for all future mutator passes.  All reverse dependencies for
+	// a destination module are collected until the end of the mutator pass, sorted by name, and
+	// then appended to the destination module's dependency list.
+	AddReverseDependency(module Module, tag DependencyTag, name string) Module
```

Comment:
This one shouldn't return the module, since modules shouldn't ever be accessing the modules that depend on them.

Reply:
Reverted changes related to `AddReverseDependency`.